packages/frontend/src/providers/TokensProvider.tsx

```diff
@@ -37,36 +44,78 @@ const TokensProvider: FunctionComponent<{ children: ReactNode }> = ({ children }
     { enabled: false }
   );
 
-  const fetchTokens = async () => {
-    const { data: tokens } = await refetch();
+  const { refetch: refetchToTokens } = useQuery(
+    ['tokens'],
```

Comment:
We now have two separate queries with the exact same key, which may cause issues. Let's do `['tokens', 'to']` for the `toTokens` query key and `['tokens', 'from']` for the `fromTokens` query key?

Reply:
Could also do `tokens-${chainId}` for caching efficiency.

Reply:
Yup, however I would do `['tokens', `${chainId}`]` in that case for a good query key hierarchy
https://tanstack.com/query/v4/docs/react/guides/query-keys#array-keys-with-variables

Reply:
Addressed in 80f0dce4f4b4e5252e84695808653f0df85daa4f. Created a hook called `useFetchTokens` to contain this logic.